### PR TITLE
added all amplify commands to spec

### PR DESCRIPTION
**Pull request by Dayton-IX**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Full spec for amplify CLI autocomplete support

**What is the current behavior? (You can also link to an open issue here)**
Only amplify env is supported

**What is the new behavior (if this is a feature change)?**
All amplify commands are supported

**Additional info:**

### CODE REVIEW

File: src/amplify.ts [hunk 1]
```diff
@@ -8,11 +8,205 @@ const envNameGenerator: Fig.Generator = {
   },
 };
 
+const amplifyCategories = [
```
Comment (QuiiBz) [inline src/amplify.ts:11]: Could you add a type of `Fig.Subcommand[]` here? And a description for each subcommand would be nice to have!

File: src/amplify.ts [hunk 2]
```diff
@@ -8,11 +8,205 @@ const envNameGenerator: Fig.Generator = {
...
+  },
+  {
+    name: "storage",
+  },
+  {
+    name: "xr",
+  },
+];
+
+const categoryCommands = [
```
Comment (QuiiBz) [inline src/amplify.ts:44]: Same as above for the type :)

File: src/amplify.ts [hunk 3]
```diff
@@ -8,11 +8,205 @@ const envNameGenerator: Fig.Generator = {
...
+    description:
+      "Removes a resource for an Amplify category in your local backend",
+  },
+  {
+    name: "push",
+    description:
+      "Provisions cloud resources with the latest local developments",
+    subcommands: [
+      {
+        name: "--y",
```
Comment (QuiiBz) [inline src/amplify.ts:66]: Is the name really `--y`, and not `-y` instead? That seems a bit strange to me

File: src/amplify.ts [hunk 4]
```diff
@@ -8,11 +8,205 @@ const envNameGenerator: Fig.Generator = {
...
+      description:
+        "Configures the attributes of your project for amplify-cli, such as switching front-end framework and adding/removing cloud-provider plugins",
+    },
+    {
+      name: "push",
+      description:
+        "Provisions cloud resources with the latest local developments",
+      subcommands: [
+        {
+          name: "--y",
```
Comment (QuiiBz) [inline src/amplify.ts:94]: Same as above

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution! I left some comments to resolve before getting this PR merged in

File: src/amplify.ts (hunk 3 above)
Comment (Dayton-IX) [reply src/amplify.ts:66]: yeah it really is --y!

**Review by QuiiBz — APPROVED** [review]
LGTM, thank you so much for this contribution!

**Pull request merged**